recipe/meta.yaml

```diff
@@ -20,38 +20,42 @@ build:
 requirements:
   host:
     - git
-    - python >=3.8
+    - python >=3.8,<3.11
     - pip !=21.3
     - pytest-runner
     - setuptools_scm
   run:
     # esmvaltool
-    - python >=3.8
-    - iris >=3.2.1
+    - python >=3.8,<3.11
+    - iris >=3.4.0
     - python-stratify
     - scipy >=1.6
     # Normally installed via pip:
     - cartopy
     - cftime
-    - cf-units >=3.0.0, <3.1.0, !=3.0.1.post0
+    - cf-units
     - dask
     - esgf-pyclient >=0.3.1
-    - esmpy !=8.1.0
+    - esmpy !=8.1.0,<8.4.0
```

Comment:
indeed this needs be kept until iris-esmf-regrid fix their stuff (I got at least one PR open there) - once this gets freed and we make prospector optional then Python=3.11 support comes as a bonus immediately :grin: